client/checkout/platform-checkout/email-input-iframe.js

```diff
@@ -244,10 +244,26 @@ export const handlePlatformCheckoutEmailInput = ( field, api ) => {
 		return pattern.test( value );
 	};
 
-	// Check the initial value of the email input and trigger input validation.
-	if ( validateEmail( platformCheckoutEmailInput.value ) ) {
-		platformCheckoutLocateUser( platformCheckoutEmailInput.value );
-	}
+	// Prevent show platform checkout iframe if the page comes from browser back button
+	// or from the back button on platform checkout itself.
+	window.addEventListener( 'pageshow', function ( event ) {
+		const historyTraversal =
+			event.persisted ||
+			( 'undefined' !== typeof performance &&
+				'back_forward' ===
+					performance.getEntriesByType( 'navigation' )[ 0 ].type );
+
+		const searchParams = new URLSearchParams( window.location.search );
+
+		if (
+			! historyTraversal &&
```

Comment:
When coming back through the browser's back button, I still see the SMS OTP modal open with the loading state. Which is pretty much the last thing displayed before getting redirected. Do you get that on your end as well?

I think that'd make sense because the page is loaded directly from the bfcache - instead of loading it normally and invoking all of the JS flow for rendering it - so I don't think this approach will solve that issue.

Because we're changing `javascript:history.back()` to the actual merchant's site URL and using history only as a fallback, we could reduce the scope of the issue we're addressing here to cover the behavior of the buttons in the UI only, and have another issue opened for handling the "back" navigation via history. Thinking that skipping the bfcache for this case might be tricky and doesn't seem to have a high priority: you can always close the modal with the X button and the current behavior doesn't seem to affect the UX, even less when the behaviors of the buttons of the UI are addressed. My 2c really. Glad to ping the ones in the conversation about the original issue to chime in.

Reply:
Got it, I've noticed that the browser back button event works in Chrome but do not work in Safari, removed it on c255b8b, I'll investigate it better and open another PR later on.